builder.go

```diff
@@ -264,6 +264,16 @@ func extractNameFromNode(node *parser.Node) (string, bool) {
 	return n.Next.Value, true
 }
 
+func argsForStage(stageRoot *parser.Node, allArgs map[string]string) map[string]string {
```

Comment:
For history's sake, it would be good to have a comment for this function talking about what you expect it to do.